client/components/number-field/number-input.js

```diff
@@ -1,44 +1,38 @@
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
 import FormTextInput from 'components/forms/form-text-input';
 
-export default React.createClass( {
-	displayName: 'NumberInput',
-
-	propTypes: {
+export default class NumberInput extends Component {
+	static propTypes = {
 		onChange: PropTypes.func,
-	},
+	};
 
-	getDefaultProps() {
-		return {
-			onChange: () => {},
-		};
-	},
+	static defaultProps = {
+		onChange: () => {},
+	};
 
-	getInitialState() {
-		return {
-			focused: false,
-			text: this.props.value,
-		};
-	},
+	state = {
+		focused: false,
+		text: this.props.value,
+	};
```

Comment:
Some of the newer syntax that's available didn't make it to the docs yet. This is the [reference](https://babeljs.io/blog/2015/06/07/react-on-es6-plus) I use when converting ES5 to ES6+.

In particular I'm using [property initializers](https://babeljs.io/blog/2015/06/07/react-on-es6-plus#property-initializers) here, which are available to WCS because we're using the [stage 1](https://github.com/Automattic/woocommerce-services/blob/master/webpack.config.js#L24) preset, which [includes](https://babeljs.io/docs/plugins/preset-stage-1/) stage 2, which currently includes [`transform-class-properties`](https://babeljs.io/docs/plugins/transform-class-properties/).